cis194/week1/zhansongl/credit_card.hs

```diff
@@ -0,0 +1,125 @@
+import Test.QuickCheck
+import Test.Hspec
+
+toDigitsRev :: Integer -> [Integer]
+toDigitsRev n
+  | n >= 0 = digit n
+  | otherwise = []
+  where digit 0 = []
+        digit n = n `mod` 10 : digit (n `div` 10)
+
+toDigitsRev_prop n = (reverse . joinDigits $ toDigitsRev n) == (showDigits n)
+  where types = n::Integer
+        joinDigits xs = foldr (++) "" $ map show xs
+        showDigits n
+          | n <= 0 = "" -- weird restriction, but okay
+          | otherwise = show n
+
+toDigits :: Integer -> [Integer]
+toDigits = reverse . toDigitsRev
+
+toDigits_prop n = (joinDigits $ toDigits n) == (showDigits n)
+  where types = n::Integer
+        joinDigits xs = foldr (++) "" $ map show xs
+        showDigits n
+          | n <= 0 = "" -- weird restriction, but okay
+          | otherwise = show n
+
+doubleEveryOtherRev :: [Integer] -> [Integer]
+doubleEveryOtherRev [] = []
+doubleEveryOtherRev [x] = [x]
+doubleEveryOtherRev (x:y:xs) = (x:2*y:doubleEveryOtherRev xs)
```

Comment:
nit: Usually, we put spaces before and after `:`. For some codebase, we even put space after `(` and before `)`.

```suggestion
doubleEveryOtherRev ( x : y : xs ) = ( x : 2 * y : doubleEveryOtherRev xs )
```

Reply:
will fix